File: 2605-form-smallest-number-from-two-digit-arrays/2605-form-smallest-number-from-two-digit-arrays.cpp
class Solution {
public:
    int minNumber(vector<int>& nums1, vector<int>& nums2) {
        vector<int> freq(10);
        for(int i=0; i<nums1.size(); i++) freq[nums1[i]]++;
        for(int i=0; i<nums2.size(); i++) freq[nums2[i]]++;
            
        int k=2;
        int ans =0;
        int mn1 = *min_element(nums1.begin(), nums1.end());
        int mn2 = *min_element(nums2.begin(), nums2.end());
        for(int i=1; i<=9; i++)
            if(freq[i]==2) return i;
        
        if(mn2<mn1) swap(mn2, mn1);
        return mn1*10+mn2;
    }
};

File: 2605-form-smallest-number-from-two-digit-arrays/2605-form-smallest-number-from-two-digit-arrays.java
class Solution {
    public int minNumber(int[] nums1, int[] nums2) {
        Arrays.sort(nums1);
        Arrays.sort(nums2);
        int a=nums1[0];
        int b=nums2[0];
        HashSet<Integer> hs1=new HashSet<>();
        HashSet<Integer> hs2=new HashSet<>();
        for(int i=0;i<nums1.length;i++){
            hs1.add(nums1[i]);
        }
         for(int i=0;i<nums2.length;i++){
            hs2.add(nums2[i]);
        }
        for(int i=0;i<nums2.length;i++){
            if(hs1.contains(nums2[i]))
                return nums2[i];
        }
        if(a==b)
            return a;
        
        if(a<b)
        return (a*10)+b;
        
        return (b*10)+a;
    }
}